src/overlays/actors/ovl_Object_Kankyo/z_object_kankyo.c

```diff
@@ -32,41 +57,879 @@ const ActorInit Object_Kankyo_InitVars = {
+                }
+            } else {
+                this->effects[0].amplitude -= 1.0f / 7.0f;
+                if (this->effects[0].amplitude <= 0.1f) {
+                    this->effects[0].amplitude = 0.0f;
+                    this->effects[0].state = 0;
+                }
+            }
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Object_Kankyo/ObjectKankyo_Destroy.s")
+            if (++this->effects[0].timer > 7) {
+                this->effects[0].timer = 0;
+            }
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Object_Kankyo/func_80BA2DC8.s")
+            start.x = globalCtx->csCtx.npcActions[1]->startPos.x;
+            start.y = globalCtx->csCtx.npcActions[1]->startPos.y;
+            start.z = globalCtx->csCtx.npcActions[1]->startPos.z;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Object_Kankyo/func_80BA2DD4.s")
+            end.x = globalCtx->csCtx.npcActions[1]->endPos.x;
+            end.y = globalCtx->csCtx.npcActions[1]->endPos.y;
+            end.z = globalCtx->csCtx.npcActions[1]->endPos.z;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Object_Kankyo/ObjectKankyo_Update.s")
+            interp = func_8006F93C(globalCtx->csCtx.npcActions[1]->endFrame,
+                                   globalCtx->csCtx.npcActions[1]->startFrame,
+                                   globalCtx->csCtx.frames);
+            Matrix_Translate((end.x - start.x) * interp + start.x,
+                             (end.y - start.y) * interp + start.y,
+                             (end.z - start.z) * interp + start.z, MTXMODE_NEW);
+            Matrix_Scale(this->effects[0].size, this->effects[0].size, this->effects[0].size, MTXMODE_APPLY);
+            func_80093D84(globalCtx->state.gfxCtx);
+            gDPPipeSync(POLY_XLU_DISP++);
+            gDPSetPrimColor(POLY_XLU_DISP++, 0, 0, (u8)(105 * this->effects[0].amplitude) + 150, 255,
+                            (u8)(105 * this->effects[0].amplitude) + 150, this->effects[0].alpha);
+            gDPSetEnvColor(POLY_XLU_DISP++, (u8)(155 * this->effects[0].amplitude) + 100,
+                           (u8)(255 * this->effects[0].amplitude), 255 - (u8)(255 * this->effects[0].amplitude),
+                           this->effects[0].alpha);
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Object_Kankyo/ObjectKankyo_Draw.s")
+            Matrix_Mult(&globalCtx->mf_11DA0, MTXMODE_APPLY);
+            gSPMatrix(POLY_XLU_DISP++, Matrix_NewMtx(globalCtx->state.gfxCtx, "../z_object_kankyo.c", 1416),
+                      G_MTX_LOAD);
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Object_Kankyo/func_80BA3C94.s")
+            gSPSegment(POLY_XLU_DISP++, 0x08, SEGMENTED_TO_VIRTUAL(D_80BA5900[this->effects[0].timer]));
+            gDPPipeSync(POLY_XLU_DISP++);
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Object_Kankyo/func_80BA414C.s")
+            gSPDisplayList(POLY_XLU_DISP++, D_06009620);
```

Comment:
```suggestion
            gSPDisplayList(POLY_XLU_DISP++, object_spot02_objects_DL_009620);
```
\+ `#include "objects/object_spot02_objects/object_spot02_objects.h"`
and remove extern/ undefined_syms.txt entry